lib/preprocessCss.js

```diff
@@ -282,8 +282,9 @@ module.exports = (app, callback) => {
                 content = fs.readFileSync(outpath, 'utf8')
               }
 
-              // check existing file for matching content before writing
-              if (!content || content !== newCSS) {
+              // check that the newCSS has content before writing the file. It is possible that the less file has no valid CSS and results in an empty CSS file. In this case don't write it.
```

Comment:
Are we sure we don't want to print the empty file? An empty file seems like a pretty good indication something is wrong with the CSS, as opposed to no file at all which one might assume is a Roosevelt problem.

Reply:
Let me give you an example. Image a file like this `colors.less`

```less
@green: #00c000;
@yellow: #ffed29;
@red: #ff6969;
@cyan: #00ced1;
@white: #fff;
```

This will never generate a `.css` file. Instead when the less is processed these values are sprinkled in the respective `.css` files. If you don't check the newCss content for empty you will write an empty file.

Every time the CSS get's processed it will diff the _old_ CSS against the _new_ CSS and it will write the blank file every time. This will also generate a log, when there is no CSS changes in the first place.

Reply:
Ah alright, that makes sense.